scarlet/bbox.py

```diff
@@ -11,22 +11,17 @@ class Box:
     Parameters
     ----------
     shape: tuple
-        Size of the box in depth,height,width
+        Size of the box
     origin: tuple
-        Minimum (z,y,x) value of the box (front low left corner).
+        Front/low/left corner coordinate of the box
     """
 
-    def __init__(self, shape, origin=(0, 0, 0)):
-        # bbox always in 3D
-        if len(shape) == 2:
-            shape = (0, *shape)
-        assert len(shape) == 3
-        self.shape = shape
-
-        if len(origin) == 2:
-            origin = (0, *origin)
-        assert len(origin) == 3
-        self.origin = origin
+    def __init__(self, shape, origin=None):
+        self.shape = tuple(shape)
+        if origin is None:
+            origin = (0,) * len(shape)
+        assert len(origin) == len(shape)
+        self.origin = tuple(origin)
```

Comment:
That's nice and cleaner, although I think I raised it at the previous PR, why would we need anything else but 3-D? Already 2D is, I think unecessary if we specify that everything has to be 3-D

Reply:
Morphology isn't 3D, so I think that it's ok to have 2D boxes, but this makes me think that there should be a way to "slice" a box to take a 3D box and extract only the 2D portion of it (which can obviously be done by hand using the `from_bounds` method but this might be nice syntactic sugar.

Reply:
We only use 2D and 3D (so far?...), however, we implicitly cast 2D boxes to 3D. that create a lot of code bulk, which this PR removes. I can think of 4D (Time, Channel, Height, Width) or so, but it's largely about code structure not functionality.

Reply:
Thanks for clearing this out. 
And yes, I can see how ND data can exist, but it would be conceptually a different regime.